src/main/scala/model/entities/Enemy.scala

```diff
@@ -2,25 +2,26 @@ package model.entities
 
 import model.common.DefaultValues
 import model.entities.TroopState.*
-import model.entities.WorldSpace.Position
+import model.entities.WorldSpace.{LanesLength, NumOfLanes, Position}
 
 import scala.concurrent.duration.FiniteDuration
+import scala.util.Random
 
 /**
- * Interface of an enemy model
+ * A Enemy is an abstract entity that models the common behaviour of different types of enemy.
+ *
+ * @param position the position in which the enemy is placed.
+ * @param life the life that the enemy currently has.
+ * @param state the state of the enemy.
  */
 abstract class Enemy(override val position: Position,
             override val life: Int,
-            override val state: TroopState,
-            override val velocity: Float) extends Troop with MovingAbility:
+            override val state: TroopState) extends Troop with MovingAbility:
 
   override def isInterestedIn: Entity => Boolean =
```

Comment:
By looking at this method it's not clear what it does.
Refactor in the same way we did in the Plant file.